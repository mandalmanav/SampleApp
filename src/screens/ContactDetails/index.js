import React from 'react'
import { View, Text, StyleSheet, Pressable, Image, Alert } from 'react-native'
import { Icon } from 'react-native-elements'
import { useDispatch, useSelector } from 'react-redux'
import { deleteContactAction } from '../../store/actions/contacts'

export default ({ route, navigation }) => {
    let { contact } = useSelector(state => state)
    const dispatch = useDispatch()
    const deleteContact = (contact) => {
      
        Alert.alert(
            "Delete",
            "Are you sure you want to delete this contact ?",
            [
                {
                    text: "Cancel",
                    onPress: () => {
                        return
                    },
                    style: "cancel"
                },
                {
                    text: "Yes, Delete", onPress: async () => {
                        try {
                            await deleteContactAction(dispatch, contact.id)
                            navigation.navigate('ContactsList')
                        }
                        catch (e) {
                            alert(e)
                        }
                    }
                }
            ]
        );

    }

    return <View style={styles.container}>
        <View style={styles.head}>

            <View style={styles.avatarBar}>
                <View style={styles.avatarBox}>
                    <Image
                        testID="image"
                        style={styles.avatar}
                        source={{
                            uri: contact.image,
                        }}
                    />
                </View>
            </View>
            <View style={styles.nameView}>
                <Text testID="full-name" style={styles.label}>{contact.firstName} {contact.lastName}</Text>
            </View>
            <View style={styles.actionBar}>
                <Icon
                    testID="comment-icon"
                    reverse
                    reverseColor='white'
                    type='font-awesome'
                    name='comment'
                    color="#4d90fe"
                    size={20}
                />
                <Icon
                    testID="phone-icon"
                    reverse
                    reverseColor='white'
                    type='font-awesome'
                    name='phone'
                    color="#4d90fe"
                    size={20}
                />
                <Icon
                    testID="camera-icon"
                    reverse
                    reverseColor='white'
                    type='font-awesome'
                    name='video-camera'
                    color="#4d90fe"
                    size={20}
                />
                <Icon
                    testID="envelope-icon"
                    reverse
                    reverseColor='white'
                    type='font-awesome'
                    name='envelope'
                    color="#4d90fe"
                    size={20}
                />
            </View>
        </View>
        <View style={styles.content}>
            <View style={styles.formItem} >
                <Text testID="mobile-label" style={styles.formLabel}>Mobile</Text>
                <Text testID="mobile" style={styles.formValue}>{contact.countryCode} {contact.mobile}</Text>
            </View>

            <View style={{ flexDirection: 'row', marginTop: 40, justifyContent: 'space-around' }}>
                <View style={{
                    alignItems: 'flex-start',
                    marginLeft: 20
                }}>
                    <Pressable testID="delete-button" onPress={() => deleteContact(contact)}>
                        <Text style={{
                            padding: 8,
                            backgroundColor: 'red',
                            fontWeight: 'bold',
                            color: '#fff'
                        }}>Delete Contact</Text>
                    </Pressable>
                </View>
               
            </View>

        </View>

    </View>
}
const styles = StyleSheet.create({
    container: {
        height: '100%'
    },

    head: {
        padding: 8,
        paddingBottom: 78
    },
    navBar: {
        flexDirection: 'row',
        justifyContent: 'space-between'


    },
    backButton: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    edit: {
        fontSize: 18,
        color: '#4d90fe'
    },
    avatarBar: {
        flexDirection: 'row',
        justifyContent: 'center'
    },
    avatar: {
        width: 50,
        height: 50,
        borderRadius: 50 / 2,
        overflow: "hidden",
    },
    avatarBox: {
        width: 70,
        height: 70,
        borderRadius: 70 / 2,
        overflow: "hidden",
        borderWidth: 1,
        borderColor: '#e5e5e5',
        alignItems: 'center',
        justifyContent: 'center'
    },
    nameView: {
        padding: 4,
        alignItems: 'center'
    },
    label: {
        fontSize: 22,
        fontWeight: 'bold',
        padding: 4
    },
    actionBar: {
        padding: 8,
        flexDirection: 'row',
        justifyContent: 'space-evenly'
    },
    content: {
        backgroundColor: '#fff',
        flex: 3
    },
    formItem: {
        backgroundColor: '#fff',
        padding: 8,
        paddingBottom: 0,

    },
    formLabel: {
        fontSize: 18
    },
    formValue: {
        fontSize: 26,
        color: '#4d90fe',
        paddingTop: 6,
        paddingBottom: 6,
        borderBottomColor: '#e7e7e7',
        borderBottomWidth: .5
    }

})