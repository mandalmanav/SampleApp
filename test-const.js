export const CONTACTS=[{
    firstName:'Manav',
    lastName:'Mandal',
    image:'',
    mobile:'9999999999',
    id:'q',
    countryCode:'+91'
},{
    firstName:'Aman',
    lastName:'Vatsa',
    image:'',
    mobile:'9999999999',
    id:'r',
    countryCode:'+91'
}]

export const SEARCH_TERM='am'