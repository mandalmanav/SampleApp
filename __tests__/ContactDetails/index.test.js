
import ContactDetails from '../../src/screens/ContactDetails'
import { render, fireEvent } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { CONTACTS, SEARCH_TERM } from '../../test-const';
import { it } from '@jest/globals';

import { Alert } from 'react-native';
jest.spyOn(Alert, 'alert');
describe('TEst Contact detail page', () => {

    let initialState = { contacts: CONTACTS, searchTerm: '',contact:CONTACTS[0] }
    const mockStore = configureStore()
    let store, wrapper
    store = mockStore(initialState)


    it('Full Name is visilble', () => {
        const { getByText } = render(<Provider store={store}><ContactDetails  /></Provider>)
        let item = getByText(`${CONTACTS[0].firstName} ${CONTACTS[0].lastName}`)
        expect(item).not.toBeNull()
    })
    it('Mobile number is visilble', () => {
        const { getByText } = render(<Provider store={store}><ContactDetails  /></Provider>)
        let item = getByText(`${CONTACTS[0].countryCode} ${CONTACTS[0].mobile}`)
        expect(item).not.toBeNull()
    })
    it('Icons are rendered', () => {
        const { getByTestId } = render(<Provider store={store}><ContactDetails  /></Provider>)

        expect(getByTestId('comment-icon')).not.toBeNull()
        expect(getByTestId('phone-icon')).not.toBeNull()
        expect(getByTestId('camera-icon')).not.toBeNull()
        expect(getByTestId('envelope-icon')).not.toBeNull()
    })
    // it('Edit button is rendered and clickable', () => {
    //     const mockCallBack = jest.fn();
    //     const { getByTestId, getByText } = render(<Provider store={store}><ContactDetails route={{
    //         params: {
    //             contact: CONTACTS[0]
    //         }
            
    //     }} navigation={
    //         {
    //             navigate:mockCallBack
    //         }
    //     }/></Provider>)
    //     expect(getByText('Edit Contact >>')).not.toBeNull()
    //     expect(getByTestId('edit-button')).not.toBeNull()
    //     fireEvent.press(getByTestId('edit-button'))
    //     expect(mockCallBack.mock.calls.length).toEqual(1);

    // })
    it('delete button is rendered and clickable', () => {
        // const mockCallBack = jest.fn();
        const { getByTestId, getByText } = render(<Provider store={store}><ContactDetails route={{
            params: {
                contact: CONTACTS[0]
            }
            
        }} /></Provider>)
        expect(getByText('Delete Contact')).not.toBeNull()
        expect(getByTestId('delete-button')).not.toBeNull()
        fireEvent.press(getByTestId('delete-button'))
        expect(Alert.alert).toHaveBeenCalled()
    })
    it('Match Snapshot', () => {
        const component = render(<Provider store={store}><ContactDetails  /></Provider>)
        expect(component).toMatchSnapshot()
    })
})