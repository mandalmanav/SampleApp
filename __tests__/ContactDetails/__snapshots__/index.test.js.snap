// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEst Contact detail page Match Snapshot 1`] = `
<View
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View
    style={
      Object {
        "padding": 8,
        "paddingBottom": 78,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#e5e5e5",
            "borderRadius": 35,
            "borderWidth": 1,
            "height": 70,
            "justifyContent": "center",
            "overflow": "hidden",
            "width": 70,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "",
            }
          }
          style={
            Object {
              "borderRadius": 25,
              "height": 50,
              "overflow": "hidden",
              "width": 50,
            }
          }
          testID="image"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "padding": 4,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 22,
            "fontWeight": "bold",
            "padding": 4,
          }
        }
        testID="full-name"
      >
        Manav
         
        Mandal
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "padding": 8,
        }
      }
    >
      <View
        style={
          Object {
            "borderRadius": 24,
            "height": 44,
            "margin": 7,
            "overflow": "hidden",
            "width": 44,
          }
        }
      >
        <View
          testID="comment-icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4d90fe",
                "borderRadius": 24,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 24,
            "height": 44,
            "margin": 7,
            "overflow": "hidden",
            "width": 44,
          }
        }
      >
        <View
          testID="phone-icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4d90fe",
                "borderRadius": 24,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 24,
            "height": 44,
            "margin": 7,
            "overflow": "hidden",
            "width": 44,
          }
        }
      >
        <View
          testID="camera-icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4d90fe",
                "borderRadius": 24,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderRadius": 24,
            "height": 44,
            "margin": 7,
            "overflow": "hidden",
            "width": 44,
          }
        }
      >
        <View
          testID="envelope-icon"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#4d90fe",
                "borderRadius": 24,
                "height": 44,
                "justifyContent": "center",
                "width": 44,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 20,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "FontAwesome",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 3,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "padding": 8,
          "paddingBottom": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
        testID="mobile-label"
      >
        Mobile
      </Text>
      <Text
        style={
          Object {
            "borderBottomColor": "#e7e7e7",
            "borderBottomWidth": 0.5,
            "color": "#4d90fe",
            "fontSize": 26,
            "paddingBottom": 6,
            "paddingTop": 6,
          }
        }
        testID="mobile"
      >
        +91
         
        9999999999
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 40,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "marginLeft": 20,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          testID="delete-button"
        >
          <Text
            style={
              Object {
                "backgroundColor": "red",
                "color": "#fff",
                "fontWeight": "bold",
                "padding": 8,
              }
            }
          >
            Delete Contact
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
